### ISSUE TITLE
 Reproducing Results from Paper

### ISSUE THREAD

**Issue by eugene6923** [diffusion-classifier/diffusion-classifier#31]
Hi! Amazing work! I've been working on reproducing the results and ran into a few things that I was hoping you could clarify when you have a moment.

1. **Winoground Compositionality Results**: I noticed that I’m getting different scores for the CLIP models on the Winoground task, even though there shouldn't be any randomness in this case. I was wondering if there might be some other settings that should be changed in the code. For reference, here are the scores I got from your code:

**CLIP ViT-L14**
Overall text score:  28.2 +/-  4.4                                   
Overall image score:  10.5 +/-  3.0                                  
Overall group score:  7.8 +/-  2.6                                   
Object text score:  27.7 +/-  7.4                                    
Object image score:  9.2 +/-  4.8                                    
Object group score:  6.4 +/-  4.0                                    
Relation text score:  25.3 +/-  5.6                                  
Relation image score:  8.6 +/-  3.6                                  
Relation group score:  5.6 +/-  2.9                                  
Both text score:  57.7 +/-  19.0                                     
Both image score:  34.6 +/-  18.3                                    
Both group score:  34.6 +/-  18.3

**Openclip H14**
Overall text score:  33.5 +/-  4.6
Overall image score:  12.8 +/-  3.3
Overall group score:  10.5 +/-  3.0
Object text score:  39.7 +/-  8.1
Object image score:  12.1 +/-  5.4
Object group score:  9.9 +/-  4.9
Relation text score:  27.0 +/-  5.7
Relation image score:  9.4 +/-  3.8
Relation group score:  7.3 +/-  3.3
Both text score:  57.7 +/-  19.0
Both image score:  46.2 +/-  19.2
Both group score:  42.3 +/-  19.0

3. **CIFAR-10 Loss Function**: In Section 5.1, it’s mentioned that "we leave the choice between ℓ1 and ℓ2 as a per-dataset inference hyperparameter." But in Table 8, I don’t see the cifar10 88.5 result that’s listed in Table 1. Could you let me know which loss function (ℓ1 or ℓ2 or hubor) was used for each dataset, and what might explain the difference between Tables 1 and 8 especially for cifar10?

Looking forward to hearing from you. Thank you!

**Comment by alexlioralexli** [comment]
1. It might be a version issue on your end? I ran the same command from my repo multiple times, and I get the same numbers as reported in the paper every time. 
2. Sorry, the number should be 88.4 using the l1 loss. I think the discrepancy is from testing with a different adaptive eval strategy.

**Comment by eugene6923** [comment]
Thank you for your response; it was very helpful! I have another question. When applying the same adaptive strategy across different datasets, I achieve similar scores, but on the Aircraft dataset, the accuracy remains around 19%. Since I noticed the seed is not fixed, do you think this could be the cause, or might there be other factors affecting the results?

**Comment by alexlioralexli** [comment]
I do consistently see around 26% on Aircraft across random seeds, so there may be some other issue on your end. It may be the same issue causing low Winoground performance for you as well.
